mne/encoding/feature.py

```diff
@@ -0,0 +1,371 @@
+"""Classes for transforming, preprocessing, and fitting encoding models."""
+# Authors: Chris Holdgraf <choldgraf@gmail.com>
+#          Jona Sassenhagen <jona.sassenhagen@gmail.de>
+#          Jean-Remi King <jeanremi.king@gmail.com>
+#
+# License: BSD (3-clause)
+import numpy as np
+from scipy.sparse import spmatrix, csr_matrix
+from ..utils import warn
+from ..decoding.mixin import TransformerMixin
+
+
+class FeatureDelayer(object):
+    """Creates delayed versions of an input stimulus.
+
+    Parameters
+    ----------
+    delays: array of floats, shape (ix_delayays,)
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means time points in the past,
+        positive means time points in the future. Default is no delays.
+    sfreq: float
+        The sampling frequency of the series. Defaults to 1.0
```

Comment:
Unnecessary param at this stage IMO; more critical to the next PR to integrate jona's usecases